modules/tooling-support/src/main/java/org/mule/runtime/module/tooling/internal/artifact/metadata/MetadataExecutor.java

```diff
@@ -73,7 +81,23 @@ protected MetadataContext createMetadataContext(Optional<ConfigurationInstance>
     return new DefaultMetadataContext(() -> configurationInstance,
                                       connectionManager,
                                       new DefaultMetadataCache(),
-                                      new JavaTypeLoader(extensionClassLoader));
+                                      new JavaTypeLoader(extensionClassLoader)) {
+
+      @Override
+      public <C> Optional<C> getConnection() throws ConnectionException {
+        long startTime = currentTimeMillis();
+        try {
+          if (LOGGER.isDebugEnabled()) {
+            LOGGER.debug("Retrieving connection");
```

Comment:
I would change the messages to something easily readable without the need to read the whole message. 
Something like: 
Get Connection STARTED for ....
...
Get Connection FINISHED for ....
 
The same for the other methods

Reply:
Done!